projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts

```diff
@@ -340,6 +340,15 @@ export class IgxHierarchicalGridComponent extends IgxHierarchicalGridBaseDirecti
         }
     }
 
+    /**
+     * @hidden
+     */
+    public get parentRowOutletDirective() {
+        // Targeting parent outlet in order to prevent hiding when outlet
+        // is present at a child grid and is attached to a row.
+        return this.parent.rowOutletDirective;
```

Comment:
Add a check if there is a parent since if this is a root grid that has <=2 rows this would throw an error.

Reply:
Thanks for noticing. It's done now.